# recursion-factorial-process-iterative
> [TABLE OF CONTENTS](https://github.com/E6985/E2746/blob/master/README.md)
***

- any problem that can be solved with - recursion - can also be solved with an - iterative implementation - preferred way to calculate a factorial in this instance as opposed to the - recursive implementation [6]

In [1]:
# define function - fFactorial
def fFactorial(nParNonNegInt):
    nResult=1
    while nParNonNegInt>1:
        nResult*=nParNonNegInt
        nParNonNegInt-=1
    return nResult

In [2]:
# call function - fFactorial
fFactorial(5)

120

- function signature - fFactorial - parameter - nParNonNegInt - five [6]

```python
def fFactorial(nParNonNegInt):
```

- nResult - assigned the value - one [6]

```python
nResult=1
```

- **while-1** - nParNonNegInt - five - greater than one - True [6]

```python
while nParNonNegInt>1:
```

- nResult - assigned - nParNonNegInt - five - times - nResult - one - 5 [6]

```python
    nResult*=nParNonNegInt
```

- decrement - nParNonNegInt - five - by one - 4 [6]

```python
    nParNonNegInt-=1
```

- **while-2** - nParNonNegInt - four - greater than one - True [6]

```python
while nParNonNegInt>1:
```

- nResult - assigned - nParNonNegInt - four - times - nResult - five - 20 [6]

```python
    nResult*=nParNonNegInt
```

- decrement - nParNonNegInt - four - by one - 3 [6]

```python
    nParNonNegInt-=1
```

- **while-3** - nParNonNegInt - three - greater than one - True [6]

```python
while nParNonNegInt>1:
```

- nResult - assigned - nParNonNegInt - three - times - nResult - twenty - 60 [6]

```python
    nResult*=nParNonNegInt
```

- decrement - nParNonNegInt - three - by one - 2 [6]

```python
    nParNonNegInt-=1
```

- **while-4** - nParNonNegInt - two - greater than one - True [6]

```python
while nParNonNegInt>1:
```

- nResult - assigned - nParNonNegInt - two - times - nResult - sixty - 120 [6]

```python
    nResult*=nParNonNegInt
```

- decrement - nParNonNegInt - two - by one [6]

```python
    nParNonNegInt-=1
```

- exit **while-5** - nParNonNegInt - one - greater than one - False [6]

```python
while nParNonNegInt>1:
```

- function - fFactorial - return - nResult - return control to caller [6]

```python
return nResult
```

***
# END